packages/flutter_tools/lib/src/web/chrome.dart

```diff
@@ -241,6 +227,66 @@ class ChromiumLauncher {
     ), skipCheck);
   }
 
+  Future<Process> _spawnChromiumProcess(List<String> args) async {
+    // Keep attempting to launch the browser until one of:
+    // - We successfully launched it, in which case we just return from the loop.
+    // - We encountered an unretriable error, in which case we throw ToolExit.
+    while (true) {
+      final Process process = await _processManager.start(args);
+
+      bool exited = false;
+      unawaited(process.exitCode.then((int exitCode) {
+        exited = true;
+      }));
+
+      process.stdout
+        .transform(utf8.decoder)
+        .transform(const LineSplitter())
+        .listen((String line) {
+          _logger.printTrace('[CHROME]: $line');
+        });
+
+      // Wait until the DevTools are listening before trying to connect. This is
+      // only required for flutter_test --platform=chrome and not flutter run.
+      //
+      // Once every few thousands of launches we hit this glibc bug: https://sourceware.org/bugzilla/show_bug.cgi?id=19329.
+      // When this happens Chrome spits out something like the following then exits with code 127:
+      //
+      //     Inconsistency detected by ld.so: ../elf/dl-tls.c: 493: _dl_allocate_tls_init: Assertion `listp->slotinfo[cnt].gen <= GL(dl_tls_generation)' failed!
+      bool hitGlibcBug = false;
+      await process.stderr
+        .transform(utf8.decoder)
+        .transform(const LineSplitter())
+        .map((String line) {
+          _logger.printTrace('[CHROME]:$line');
+          if (line.contains('Inconsistency detected by ld.so')) {
```

Comment:
Consider hoisting this into a constant, along with the documentation you added above.

Reply:
Done.